### DISCUSSION TITLE
How to verify that all photos are in the new folder?

### DISCUSSION THREAD

**Discussion by hidetheweed** [TheLastGimbus/GooglePhotosTakeoutHelper#207] [I NEED HELP]
Hi and thank you for this awesome script! So easy to use!

After running the programme successfully I now have two folders; the original folder where all my unmerged google photos went and the new "output" folder where I should find all the merged photos. 
1. Does the original unmerged folder still contain all photos or is it that when photos are merged they are moved and not copied over to the new output folder?
2. How can I check that all my photos from the unmerged folder are now in the new output folder? I see that the original unmerged folder is about 14 GB and contains approx 22 000 items. But the new merged folder is about 50 GB (!) and contains only 17 000 items. Why the mismatch?
3. Approx 6000 photos could not be merged ("couldnt find date for 6558 photos/videos"). Are these photos in the new output folder or since they have no metadata they are still in the original unmerged folder? 

I hope my questions made sense, but please do contact me if I should clarify them :) Hope someone can answer.

**Comment by TheLastGimbus** [comment]
Gpth doesn't really merge photos (with their metadata or something) - it turns out that majority of them have this correct, so gpth just nicely moves stuff and changes the "last modified" values

The default behavior is moving files, so the original input folder will be messy and have some photos moved and some not. Best is to delete it 👍

Output folder is bigger than input?? Very weird. Please try to look into this carefully and say what you found

Date not found photos should be in separate folder in the output. If you paste me the screenshot of their names, i may fix it so that they are found too

**Reply by hidetheweed** [reply]
Yes, for some reason my mac says that the input folder is approx 14GB and the new output folder is 50 GB. The input folder is called "takeout" and in there are all the unzipped takeout folders (takeout1, takeout 2, takeout 3 etc). You write that the default is behavior is MOVING files, so that might explain the mismatch in size? So some photos from the input folder were moved to the output folder and some were not. If they would have been copied it would have been easier to check that all photos from input folder went to output?

The new output folder is called "takeout with metadata" and contains a folder called "all_photos", "archive" and google albums (folders I made in google photos). All photos folders is approx 43 GB and as far as I can see does not contain any folder with photos with "date not found", so not sure where those 6000 photos went.

I think I did figure out the mismatch in items (old input had approx 22 000 items and new has only 17 000), its because mac counts folders as items as well. So if the input folder had many more folders within (as it seems), mac will show more items than the new output folder. But it still does absolutely not explain the mismatch in size...

I guess my bigges question now is: HOW can I collect all photos that came with the google takeout? Right now I'm not sure if all photos are in the output folder - I suspect some went there and some stayed in the input folder? I dont dare to delete input fodler in case there are photos that are not in the output folder.

**Reply by TheLastGimbus** [reply]
I'm back with some fresh ideas:

- the mismatch in size could be made like this:
  
  Photos takeout contains "year folders" and "album folders" - albums contain duplicates of photos that are in "year folders" anyway - `gpth` is smart and doesn't copy/move stuff when it doesn't have to

  Second thing - videos are wayy bigger than images - one minute video can weight as much as 100 photos

  Giving these two things - you may have had many albums with many photos, which duplicates of gpth left behind, while not as many videos, which makes a big difference in size :brain:

- > as far as I can see does not contain any folder with photos
  
  That's not right... :/ - check again, it really should be there... otherwise we have a dangerous bug...

  EDIT: Shit... https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper/blob/959b78b0a3c2e926c159ba67498a83ffaa6b39f5/lib/moving.dart#L105-L118

  Looks like currently it places them in "date-unknown" only when using `--divide-to-dates` - otherwise (your case) - they will just be in `ALL_PHOTOS` - i should probably change this, right?

  Anyway, your photos should be there, they will just show up at the top if you sort by modifiaction date

- > I dont dare to delete input fodler in case there are photos that are not in the output folder.
  
  Input folder is already messy, you should keep a copy of the original zip... otherwise, just re-request it if you can
  
TL;DR all your photos are in the output (99% sure) - just not all in order
